src/facade/dragonfly_connection.cc

```diff
@@ -314,6 +314,9 @@ Connection::Connection(Protocol protocol, util::HttpListenerBase* http_listener,
 
   migration_enabled_ = absl::GetFlag(FLAGS_migrate_connections);
 
+  // Create dummy value for valid control block and then use aliasing contrutor to return `this`
+  self_ = {make_shared<std::nullptr_t>(nullptr), this};
```

Comment:
1. I want it to make it for all use cases so Yue can use it too
2. Nope, https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr, it's as old as shared_ptr itself

https://stackoverflow.com/questions/27109379/what-is-shared-ptrs-aliasing-constructor-for

Briefly it means that we always return one pointer even though we manage another. As there is no real pointer for us to "manage" and we just need the atomic use-counter mechanism. 

I agree it might be confusing to understand. It's equivalent to this

```c++
struct Pointer {
  struct ControlBlock { atomic use_count; nullptr_t obj; }

  Connection* ptr;  // not managed
  ControlBlock* block;
}
```

Alternatively I can use `shared_ptr<Connection*>` without aliasing (literally heap storing a pointer to ourselves) or `shared_ptr<bool> active; ConnectionContext* self;` (where the bool makes no sense)

Or implement a custom with intrusive_ptr...

Reply:
I think with monostate it's better and more clear, nullptr_t looks scary

```c++
// Create shared_ptr that stores and empty value 
// and associate it with `this` pointer (aliasing constructor)
// We only make use of its control block for weak/strong reference counting
self_ = {make_shared<std::monostate>({}), this};
```